nilearn/decoding/space_net.py

```diff
@@ -0,0 +1,1242 @@
+"""
+sklearn-compatible implementation of spatially structured learners (
+TV-L1, S-LASSO, etc.)
+
+"""
+# Author: DOHMATOB Elvis Dopgima,
+#         PIZARRO Gaspar,
+#         VAROQUAUX Gael,
+#         GRAMFORT Alexandre,
+#         EICKENBERG Michael,
+#         THIRION Bertrand
+# License: simplified BSD
+
+import warnings
+import numbers
+import time
+import sys
+from functools import partial
+
+import numpy as np
+from scipy import stats, ndimage
+
+from sklearn.base import RegressorMixin, clone
+from sklearn.utils.extmath import safe_sparse_dot
+from sklearn.linear_model.base import LinearModel
+from sklearn.feature_selection import (SelectPercentile, f_regression,
+                                       f_classif)
+from sklearn.externals.joblib import Memory, Parallel, delayed
+from sklearn.cross_validation import check_cv
+from ..input_data import NiftiMasker
+from .._utils.fixes import center_data, LabelBinarizer, atleast2d_or_csr
+from .objective_functions import _unmask
+from .space_net_solvers import (tvl1_solver, smooth_lasso_logistic,
+                                smooth_lasso_squared_loss)
+
+
+# Volume of a standard (MNI152) brain mask in mm^3
+MNI152_BRAIN_VOLUME = 1827243.
+
+
+def _get_mask_volume(mask):
+    """Computes the volume of a brain mask in mm^3
+
+    Parameters
+    ----------
+    mask : nibabel image object
+        Input image whose voxel dimensions are to be computed.
+
+    Returns
+    -------
+    vol : float
+        The computed volume.
+    """
+    vox_dims = mask.get_header().get_zooms()[:3]
+    return 1. * np.prod(vox_dims) * mask.get_data().astype(np.bool).sum()
+
+
+def _crop_mask(mask):
+    """Crops input mask to produce tighter (i.e smaller) bounding box with
+    the same support (active voxels)"""
+    idx = np.where(mask)
+    i_min = max(idx[0].min() - 1, 0)
+    i_max = idx[0].max()
+    j_min = max(idx[1].min() - 1, 0)
+    j_max = idx[1].max()
+    k_min = max(idx[2].min() - 1, 0)
+    k_max = idx[2].max()
+    return mask[i_min:i_max + 1, j_min:j_max + 1, k_min:k_max + 1]
+
+
+def _univariate_feature_screening(
+        X, y, mask, is_classif, screening_percentile, smooth=2.):
```

Comment:
Really useful function. A few suggestions:
- we should maybe put it into _utils, I would use it for the `Decoder` class
- should it be called check_univariate_screening instead?
- should we create a _utils/checkers.py files for this?
- could the `screening_percentile` argument just be called `screening`? In the classes constructors as well?
- this would integrate nicely with a `compressing` argument for clustering methods in future averaging models for example